src/background/telemetry/exception-telemetry-listener.ts

```diff
@@ -0,0 +1,183 @@
+                }
+            } finally {
+                consoleHookIsActive = false;
+            }
+        };
+
+        // Catch errors written to logger.error
+        // Note that this is separate from console.error (despite the default logger deferring to
+        // console.error) because we generated the default logger before updating console.error
+        // above, meaning logger.error still has the old console.error implementation.
+        const loggerError = logger.error;
+        logger.error = function (message?: any, ...optionalParams: any[]) {
+            if (loggingHookIsActive) {
+                return;
+            }
+            loggingHookIsActive = true;
+            try {
+                const err = optionalParams[0] as Error;
+                if (message || err) {
+                    sendExceptionTelemetry(ErrorType.LoggerError, message, err?.stack);
+                }
+
+                if (optionalParams.length > 1) {
+                    loggerError(message, ...optionalParams);
+                } else {
+                    loggerError(message, err);
+                }
+            } finally {
+                loggingHookIsActive = false;
+            }
+        };
+    }
+
+    private sendExceptionTelemetry = (
+        errorType: string,
+        message: string,
+        stackTrace?: string,
+        source?: string,
+    ): void => {
+        const telemetry: UnhandledExceptionTelemetryData = {
+            message: message.toString(),
+            stackTrace,
+            source,
+            errorType,
+        };
+        const sanitizedTelemetry = this.sanitizeTelemetryData(telemetry);
+
+        if (sanitizedTelemetry) {
+            const payload: BaseActionPayload = {
+                telemetry: sanitizedTelemetry,
+            };
+
+            this.telemetryEventHandler.publishTelemetry(
+                TelemetryEvents.UNHANDLED_EXCEPTION,
+                payload,
+            );
+        }
+    };
+
+    private sanitizeTelemetryData = (
+        telemetryData: UnhandledExceptionTelemetryData,
+    ): UnhandledExceptionTelemetryData => {
+        if (telemetryData.message && telemetryData.message.length > this.MAX_MESSAGE_CHARS) {
+            telemetryData.message = telemetryData.message.substring(0, this.MAX_MESSAGE_CHARS);
+        }
+        if (telemetryData.stackTrace && telemetryData.stackTrace.length > this.MAX_STACK_CHARS) {
+            telemetryData.stackTrace = telemetryData.stackTrace.substring(0, this.MAX_STACK_CHARS);
+        }
+        if (
+            (telemetryData.message && telemetryData.message.includes('http')) ||
+            (telemetryData.stackTrace && telemetryData.stackTrace.includes('http'))
+        ) {
+            return undefined;
+        }
+        if (
+            (telemetryData.message &&
+                this.EXCLUDED_PROPERTIES.some(property =>
+                    telemetryData.message.includes(property),
```

Comment:
I think this is probably over-excluding; I'd recommend something that looks specifically for quoted versions of the excluded properties (indicating a JSON-serialized object using those props as a key), rather than also picking up error messages that just happen to include those words.

It would probably be easiest to build up a regex to do that. It would probably be most performant to try to compile the regex just one statically, along these lines:

```ts
function makeQuestionableSubstringRegex(): RegExp {
    const urlPattern = ':\\/\\/';
    const questionablePropertyNamePattern =
        this.EXCLUDED_PROPERTIES.map(escapeRegExp).join('|');
    const questionablePropertyPattern = `['"](${questionablePropertyNamePattern})['"]`;
    const questionableSubstringPattern = `${urlPattern}|${questionablePropertyPattern}`;

    // This argument *is* a constant built from literals, it's just built up from parts
    // eslint-disable-next-line security/detect-non-literal-regexp
    return new RegExp(questionableSubstringPattern);
}
```

This is complicated enough that it probably warrants moving the sanitization logic into its own independently testable component